packages/pie-toolbox/src/code/editable-html/editor.jsx

```diff
@@ -420,18 +421,74 @@ export class Editor extends React.Component {
   };
 
   onEditingDone = () => {
-    if (this.state.isHtmlMode) {
+    const { isHtmlMode, dialog, value, pendingImages } = this.state;
+
+    // Handling HTML mode and dialog state
+    if (isHtmlMode && !dialog?.open) {
```

Comment:
Here you have 2 statements that are checking isHtmlMode. Do you think we can merge them?
Like

  if (isHtmlMode) {
    // Early return if HTML mode is enabled
      if (dialog?.open) return;

      const currentValue = htmlToValue(value.document.text);
      const previewText = this.renderPreviewText(currentValue.document.text);

      this.openDialogWithPreview(currentValue, previewText);
      return;
    }

Reply:
sure, thank you!